doc/srv6/srv6_path_tracing_midpoint.md

```diff
@@ -0,0 +1,438 @@
+
+## Architecture Design 
+
+This document describes the changes required to support the PT Midpoint in SONiC.
+
+The overall architecture of SONiC is not modified. Supporting the PT Midpoint functionality in SONiC does not require any additional components.
+
+The only requirement is to extend some components (namely SONiC CLI and OrchAgent) to support the configuration and validation of port parameters required by the PT Midpoint functionality.
+
+## High-Level Design 
+
+The following diagram shows the steps required to configure the PT Midpoint feature in SONiC:
+<br> <br>
+
+![SRv6 Path Tracing Midpoint Sequence Diagram](images/srv6_path_tracing_midpoint_sequence_diagram.png "SRv6 Path Tracing Midpoint Sequence Diagram").
+
+(1) The user configures the *PT Interface ID* and *PT Timestamp Template* of a physical port via the SONiC CLI.
+
+(2) The SONiC CLI uses the _portconfig_ script to inject the PT Interface ID and PT Timestamp Template into the PORT table in CONFIG_DB.
+
+(3) portmgrd receives a PORT update notification (portmgrd is an CONFIG_DB subscriber).
+
+(4) portmgrd inject the PT Interface ID and PT Timestamp Template into the PORT_TABLE in APPL_DB.
+
+(5) PortsOrch receives a PORT_TABLE update notification (PortsOrch is an APPL_DB subscriber).
+
+(6) PortsOrch invokes the sairedis `set_port_attribute()` APIs to set the port PT attributes including PT Interface ID, PT Timestamp Template, PT TAM Object. The port PT attributes are injected  into the ÀSIC_DB.
+
+The next subsections describe the changes required to support the PT Midpoint HLD described above. 
+
+### SONiC CLI Changes
+
+#### CLI Configuration Commands
+
+The existing `config interface` command is extended by introducing a new subcommand `pt` that allows users to configure the Interface ID and Timestamp Template parameters required for the PT Midpoint functionality. The Timestamp Template is optional. The default template is *template3*.
+
+```
+admin@sonic:~# config interface pt <interface_name> <pt_interface_id> [pt_timestamp_template]
+
+    <interface_name>: interface name 
+    <pt_interface_id>: Path Tracing Interface ID (0-4095)
+    [pt_timestamp_template]: Path Tracing timestamp template (optional)
+        Supported templates {template1, template2, template3, template4}
+            template1: timestamp[08:15]
+            template2: timestamp[12:19]
+            template3: timestamp[16:23]
+            template4: timestamp[20:27]
+        Default: template3
+```
+
+**Example 1:**
+
+```
+admin@sonic:~# config interface pt Ethernet8 128
+```
+
+The above command assigns Ethernet8 a Path Tracing Interface ID of 128 and a Timestamp Template template3 (default).
+
+**Example 2:**
+
+```
+admin@sonic:~# config interface pt Ethernet9 129 template2
+```
+
+The above command assigns Ethernet9 a Path Tracing Interface ID of 129 and a Timestamp Template template2.
+
+#### CLI Show Commands
```

Comment:
please define the behavior when the underlying sai attribute is not support. we should query from sai and get the capability, put into state db, and cli will check the state db flag first, if not supported, then return error to the user.